app/org/sagebionetworks/bridge/services/UploadSchemaService.java

```diff
@@ -187,4 +208,39 @@ public UploadSchema getUploadSchemaByIdAndRev(StudyIdentifier studyIdentifier, S
     public List<UploadSchema> getUploadSchemasForStudy(StudyIdentifier studyId) {
         return uploadSchemaDao.getUploadSchemasForStudy(studyId);
     }
+
+    /**
+     * Service handler for updating schema revision, using V4 API semantics. See
+     * {@link UploadSchemaDao#updateSchemaRevisionV4}
+     *
+     * @param studyId
+     *         study that contains the schema
+     * @param schemaId
+     *         schema ID to update
+     * @param schemaRevision
+     *         schema revision to update
+     * @param uploadSchema
+     *         schema with updates to persist
+     * @return the updated schema revision
+     */
+    public UploadSchema updateSchemaRevisionV4(StudyIdentifier studyId, String schemaId, int schemaRevision,
+            UploadSchema uploadSchema) {
+        // Validate inputs. Study ID is the only one that doesn't need validation, since it comes from the controller,
+        // not user input.
+        if (StringUtils.isBlank(schemaId)) {
+            throw new BadRequestException("Schema ID must be specified");
+        }
+
+        if (schemaRevision <= 0) {
+            throw new BadRequestException("Schema revision must be positive");
+        }
+
+        if (uploadSchema == null) {
+            throw new InvalidEntityException("Upload schema can't be null");
+        }
```

Comment:
Fixed for uploadSchema. Kept for schemaId and schemaRev because user can submit invalid values for those.